jetty-core/jetty-http2/jetty-http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/HpackDecoder.java

```diff
@@ -138,9 +140,18 @@ public MetaData decode(ByteBuffer buffer) throws HpackException.SessionException
                 {
                     if (LOG.isDebugEnabled())
                         LOG.debug("decode Idx {}", entry);
+
+                    String name = field.getName();
+                    if (!HttpTokens.isLegalH2H3FieldName(name))
+                        _builder.streamException("Illegal header name %s", name);
```

Comment:
Should this be 
```suggestion
                    String name = field.getName();
                    if (field.getHeader() == null && !HttpTokens.isLegalH2H3FieldName(name))
                        _builder.streamException("Illegal header name %s", name);
```

That will be faster in the most common case.  We could have a lowercase only cache lookup (at line 223 below ) for finding known fields in the first place?